src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java

```diff
@@ -141,5 +145,74 @@ public FormValidation doValidateForm(
         ) {
             return MablStepBuilderValidator.validateForm(restApiKey, environmentId, applicationId);
         }
+
+        public ListBoxModel doFillApplicationIdItems(@QueryParameter String restApiKey) {
+            if(restApiKey == null || restApiKey.isEmpty()) {
+                ListBoxModel items = new ListBoxModel();
+                items.add("Input an Api Key", "");
+
+                return items;
+            }
+
+            return getApplicationIdItems(restApiKey);
+        }
+
+        private ListBoxModel getApplicationIdItems(String formApiKey) {
+            final MablRestApiClient client = new MablRestApiClientImpl(MABL_REST_API_BASE_URL, formApiKey);
+            ListBoxModel items = new ListBoxModel();
+            try {
+                GetApiKeyResult apiKeyResult = ((MablRestApiClientImpl) client).getApiKeyResult(formApiKey);
+                String organizationId = apiKeyResult.organization_id;
+                GetApplicationsResult applicationsResult = ((MablRestApiClientImpl) client)
+                        .getApplicationsResult(organizationId);
+
+                items.add("","");
+                for(GetApplicationsResult.Application application : applicationsResult.applications) {
+                    items.add(application.name, application.id);
+                }
+
+                return items;
+            } catch (IOException e) {
+            } catch (MablSystemError e) {
+            }
+
+            items.add("Input a valid ApiKey", "");
+            return items;
+        }
+
+        public ListBoxModel doFillEnvironmentIdItems(@QueryParameter String restApiKey) {
+            if(restApiKey == null || restApiKey.isEmpty()) {
+                ListBoxModel items = new ListBoxModel();
+                items.add("Input an Api Key", "");
+
+                return items;
+            }
+
+            return getEnvironmentIdItems(restApiKey);
+        }
+
+        private ListBoxModel getEnvironmentIdItems(String formApiKey) {
+            final MablRestApiClient client = new MablRestApiClientImpl(MABL_REST_API_BASE_URL, formApiKey);
+            ListBoxModel items = new ListBoxModel();
+            try {
+                GetApiKeyResult apiKeyResult = ((MablRestApiClientImpl) client).getApiKeyResult(formApiKey);
+                String organizationId = apiKeyResult.organization_id;
+                GetEnvironmentsResult environmentsResult = ((MablRestApiClientImpl) client)
```

Comment:
Shouldn't need to cast to the impl. Can just expose as another method on the client interface.